lib/pbench/server/api/__init__.py

```diff
@@ -89,6 +90,11 @@ def register_endpoints(api, app, config):
         f"{base_uri}/datasets/metadata/<string:dataset>",
         resource_class_args=(config, logger),
     )
+    api.add_resource(
+        DatasetsInventory,
+        f"{base_uri}/datasets/inventory/<string:dataset>/<path:path>",
```

Comment:
In this case, `path` is always relative to the "root of the dataset", so, if the leading slash isn't present it can and should be assumed.  (But, yeah, I would like to avoid the `//` situations.)

> So I think what we want here is f"{base_uri}/datasets/inventory/<string:dataset><path:path>" ... assuming that Flask can handle that, at least, which isn't entirely clear from the documentation.

I don't see how it could handle that...how would it know where `dataset` ends and `path` starts?  I think that URI parameters have to be separated by slashes...which isn't a problem in this case because we can either assume the leading slash on `path` or glue it in there during processing.

Reply:
Yeah, you're likely right; in that case I believe there's Flask voodoo to map a missing parameter to a default instead of requiring the `//` for root, but it seems a bit ugly.

Right now on Siddardh's server, for example, `<dataset>/` returns 404 "The requested URL was not found on the server." (However, so does `//`, presumably meaning that the doubled slash isn't even enough to get Flask to process a `path` type parameter regardless of our aesthetic concerns!)